iidm/iidm-serde/src/main/java/com/powsybl/iidm/serde/extensions/BatteryShortCircuitSerDe.java

```diff
@@ -8,43 +8,96 @@
 package com.powsybl.iidm.serde.extensions;
 
 import com.google.auto.service.AutoService;
-import com.powsybl.commons.extensions.AbstractExtensionSerDe;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSortedSet;
 import com.powsybl.commons.extensions.ExtensionSerDe;
 import com.powsybl.commons.io.DeserializerContext;
 import com.powsybl.commons.io.SerializerContext;
 import com.powsybl.iidm.network.Battery;
 import com.powsybl.iidm.network.extensions.BatteryShortCircuit;
 import com.powsybl.iidm.network.extensions.BatteryShortCircuitAdder;
+import com.powsybl.iidm.serde.IidmVersion;
+import com.powsybl.iidm.serde.NetworkDeserializerContext;
+import com.powsybl.iidm.serde.NetworkSerializerContext;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.Objects;
 
 /**
  * @author Coline Piloquet {@literal <coline.piloquet@rte-france.fr>}
  */
 @AutoService(ExtensionSerDe.class)
-public class BatteryShortCircuitSerDe extends AbstractExtensionSerDe<Battery, BatteryShortCircuit> {
+public class BatteryShortCircuitSerDe extends AbstractVersionableNetworkExtensionSerDe<Battery, BatteryShortCircuit> {
 
     public BatteryShortCircuitSerDe() {
-        super("batteryShortCircuit", "network", BatteryShortCircuit.class,
-            "batteryShortCircuit.xsd", "http://www.itesla_project.eu/schema/iidm/ext/battery_short_circuit/1_0",
-            "bsc");
+        super(BatteryShortCircuit.NAME, BatteryShortCircuit.class, "bsc",
+                new ImmutableMap.Builder<IidmVersion, ImmutableSortedSet<String>>()
+                        .put(IidmVersion.V_1_0, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_1, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_2, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_3, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_4, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_5, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_6, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_7, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_8, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_9, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_10, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_11, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_12, ImmutableSortedSet.of("0.1"))
+                        .put(IidmVersion.V_1_13, ImmutableSortedSet.of("0.1", "1.0"))
+                        .build(),
+                new ImmutableMap.Builder<String, String>()
+                        .put("0.1", "http://www.itesla_project.eu/schema/iidm/ext/battery_short_circuits/1_0")
```

Comment:
As the namespace URI gives a 1.0 reference, shouldn't we call this the 1.0, and the other one 1.1 instead? Or, another way could be to call it something like 1.0-legacy?